include/boost/basic_any.hpp

```diff
@@ -0,0 +1,440 @@
+                case Typeinfo:
+                    return const_cast<void*>(static_cast<const void*>(&boost::typeindex::type_id<ValueType>().type_info()));
+            }
+
+            return 0;
+        }
+
+        template <typename ValueType>
+        static void* large_manager(operation op, basic_any& left, const basic_any* right, const boost::typeindex::type_info* info)
+        {
+            switch (op)
+            {
+                case Destroy:
+                    delete static_cast<ValueType*>(left.content.large_value);
+                    break;
+                case Move:
+                    left.content.large_value = right->content.large_value;
+                    left.man = right->man;
+                    const_cast<basic_any*>(right)->content.large_value = 0;
+                    const_cast<basic_any*>(right)->man = 0;
+                    break;
+                case Copy:
+                    left.content.large_value = new ValueType(*static_cast<const ValueType*>(right->content.large_value));
+                    left.man = right->man;
+                    break;
+                case AnyCast:
+                    return boost::typeindex::type_id<ValueType>() == *info ?
+                            static_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(left.content.large_value) : 0;
+                case UnsafeCast:
+                    return reinterpret_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(left.content.large_value);
+                case Typeinfo:
+                    return const_cast<void*>(static_cast<const void*>(&boost::typeindex::type_id<ValueType>().type_info()));
+            }
+
+            return 0;
+        }
+
+
+        template <typename ValueType>
+        struct is_small_object_impl : boost::integral_constant<bool, sizeof(ValueType) <= OptimizeForSize &&
+            boost::alignment_of<ValueType>::value <= OptimizeForAlignment &&
+            boost::is_nothrow_move_constructible<ValueType>::value>
+        {};
+
+        template <typename ValueType>
+        struct is_small_object : is_small_object_impl<ValueType>
+        {};
+
+        template <typename ValueType, typename DecayedType = BOOST_DEDUCED_TYPENAME boost::decay<const ValueType>::type>
+        static void create(basic_any& any, const ValueType& value, boost::true_type)
+        {
+            any.man = &small_manager<DecayedType>;
+            new (&any.content.small_value) ValueType(value);
+        }
+
+        template <typename ValueType, typename DecayedType = BOOST_DEDUCED_TYPENAME boost::decay<const ValueType>::type>
+        static void create(basic_any& any, const ValueType& value, boost::false_type)
+        {
+            any.man = &large_manager<DecayedType>;
+            any.content.large_value = new DecayedType(value);
+        }
+
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
```

Comment:
You have used `std::move` at line 66, so the code would not work on pre-C++11 compilers. I'd recommend concentrate on C++11 and above standards. So just assume that all the `BOOST_NO_CXX11_*` macro are not defined. Just remove the checks, simplify the code :)

Reply:
As we discussed for potential future replacement of boost::any, let's keep c++98 workability. Also fixed the "default template arguments for functions" issue